be/src/storage/rowset/bloom_filter_index_writer.cpp

```diff
@@ -225,11 +220,16 @@ class NgramBloomFilterIndexWriterImpl<TYPE_VARCHAR> : public BloomFilterIndexWri
 
                 // add this ngram into set
                 if (_values.find(unaligned_load<CppType>(&cur_ngram)) == _values.end()) {
-                    // why deep copy?
-                    _values.insert(get_value<TYPE_VARCHAR>(&cur_ngram, this->_typeinfo, &this->_pool));
+                    if (this->_bf_options.case_sensitive) {
+                        _values.insert(get_value<TYPE_VARCHAR>(&cur_ngram, this->_typeinfo, &this->_pool));
+                    } else {
+                        // todo::exist two copy of ngram, need to optimize
+                        std::string lower_ngram;
+                        Slice lower_ngram_slice = cur_ngram.tolower(lower_ngram);
+                        _values.insert(get_value<TYPE_VARCHAR>(&lower_ngram_slice, this->_typeinfo, &this->_pool));
```

Comment:
to_lower_case can be vectorized efficiently, suggest to reference to StringFunction::tolower